Code/Framework/AzFramework/AzFramework/Quality/QualityCVarGroup.cpp

```diff
@@ -0,0 +1,322 @@
+        : m_name(name)
+        , m_path(path)
+        , m_numQualityLevels(0)
+        , m_qualityLevel(QualityLevel::LevelFromDeviceRules)
+    {
+        if (auto registry = AZ::SettingsRegistry::Get(); registry != nullptr)
+        {
+            using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
+            using VisitResponse = AZ::SettingsRegistryInterface::VisitResponse;
+
+            // optional default quality level
+            registry->GetObject(m_qualityLevel, FixedValueString::format("%s/Default",
+                m_path.c_str()));
+
+            // optional description
+            registry->Get(m_description, FixedValueString::format("%s/Description",
+                m_path.c_str()));
+
+            // count the number of defined quality levels for error checking
+            AZ::SettingsRegistryVisitorUtils::VisitArray(*registry,
+                [&]([[maybe_unused]] const auto& visitArrayArg)
+                {
+                    m_numQualityLevels++;
+                    return VisitResponse::Continue;
+                },
+                FixedValueString::format("%s/Levels", m_path.c_str()));
+        }
+
+        m_functor = AZStd::make_unique<QualityCVarGroupFunctor>(
+            m_name.c_str(),
+            m_description.c_str(),
+            AZ::ConsoleFunctorFlags::DontReplicate,
+            AZ::AzTypeInfo<QualityLevel>::Uuid(),
+            *this,
+            &QualityCVarGroup::CvarFunctor);
+    }
+
+    QualityCVarGroup::~QualityCVarGroup() = default;
+
+    inline void QualityCVarGroup::CvarFunctor(const AZ::ConsoleCommandContainer& arguments)
+    {
+        StringToValue(arguments);
+    }
+
+    QualityLevel QualityCVarGroup::GetQualityLevel(const AZ::CVarFixedString& value) const
+    {
+        // We don't use AZ::ConsoleTypeHelpers::ToValue because it will print
+        // a warning when it fails to convert a string to int32_t
+        // The QualityLevel strings are user-defined and not known at compile
+        // time so only the pre-defined values can be converted.
+
+        // handle integer values (fast path)
+        QualityLevel qualityLevel = FromNumber(value);
+
+        if (qualityLevel == QualityLevel::Invalid)
+        {
+            // handle enum pre-defined values 
+            qualityLevel = FromEnum(value);
+        }
+
+        if (qualityLevel == QualityLevel::Invalid)
+        {
+            // handle user-defined level name values
+            qualityLevel = FromName(value);
+        }
+
+        return qualityLevel;
+    }
+
+    QualityLevel QualityCVarGroup::FromNumber(const AZ::CVarFixedString& value) const
+    {
+        char* endPtr = nullptr;
+        int32_t intValue = aznumeric_cast<int32_t>(strtoll(value.c_str(), &endPtr, 0));
+
+        // don't allow values higher than the user-defined levels
+        if (endPtr == value.c_str() || intValue >= m_numQualityLevels)

```

Comment:
It is recomemmed to use the C++17 from_chars going forward to convert integers.
We have examples in the [Charconv.cpp](https://github.com/o3de/o3de/blob/ef63f6fa8ffb30027a7bdc3f4d349766531c6d19/Code/Framework/AzCore/Tests/AZStd/Charconv.cpp#L175-L178) unit test
```suggestion
        int32_t intValue;
        AZStd::from_chars_result result = AZStd::from_chars(value.begin(),
            value.end(), intValue);

        // don't allow values higher than the user-defined levels
        if (result != AZStd::errc{} || intValue >= m_numQualityLevels)

```